src/foam/dao/jdbc/MySQLJDBCDAO.java

```diff
@@ -0,0 +1,333 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao.jdbc;
+
+import foam.core.*;
+
+import foam.dao.Sink;
+
+import foam.nanos.logger.Logger;
+
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+
+import java.sql.*;
+
+import java.sql.SQLException;
+import java.util.*;
+
+/**
+  We assume that the database is created by a script when the system starts the first time.
+  When the system restarts at any point in time it should verify that the database is already created otherwise it should create it.
+  When creating the database, we only do a CREATE SCHEMA sql instruction, then we create the app user and password and grant him full privileges on this database.
+  Any other database objects (tables) will be created on the fly by the application (when methods of this class are called).
+
+ */
+public class MySQLJDBCDAO extends AbstractJDBCDAO{
+
+  public MySQLJDBCDAO(X x, ClassInfo of, String poolName) throws java.sql.SQLException, ClassNotFoundException {
+    super(x, of, poolName);
+  }
+
+  @Override
+  public FObject put_(X x, FObject obj) {
```

Comment:
Is there anything MySQL specific about this put_() method? Could it be moved to AbstractJDBDCDAO?

Reply:
yes, the select options are specific to MySQL ( when the object is already there, the update part that was wrote in postgresDAO does not comply with MySQL syntax)